docs/smart-contracts/guides/transactions/simulateTransaction-Deep-Dive.mdx

```diff
@@ -0,0 +1,531 @@
+        try {
+          const response = await sendTransaction(signedTx, (result) => {
+            console.log(`bump ttl for contract result: ${result}`);
+            return result;
+          });
+          return response;
+        } catch (error) {
+          console.error("Transaction submission failed with error:", error);
+          throw error;
+        }
+      }
+    } else {
+      console.log("No ledger entry found for the given WASM hash.");
+    }
+  } catch (error) {
+    console.error("Failed to simulate restoration:", error);
+    throw error;
+  }
+  return undefined;
+}
+
+/**
+ * Handles the restoration of a Soroban contract.
+ * @param {SorobanRpc.Api.SimulateTransactionRestoreResponse} simResponse - The simulation response containing restoration information.
+ * @param {TxParams} txParams - The transaction parameters.
+ * @returns {Promise<void>} A promise that resolves when the restoration transaction has been submitted.
+ */
+export async function handleRestoration(
+  simResponse: SorobanRpc.Api.SimulateTransactionRestoreResponse,
+  txParams: TxParams,
+): Promise<void> {
+  const restorePreamble = simResponse.restorePreamble;
+  console.log("Restoring for account:", txParams.account.accountId());
+  const account = await config.rpc.getAccount(txParams.account.accountId());
+  // Construct the transaction builder with the necessary parameters
+  const restoreTx = new TransactionBuilder(account, {
+    ...txParams.txBuilderOptions,
+    fee: restorePreamble.minResourceFee, // Update fee based on the restoration requirement
+  })
+    .setSorobanData(restorePreamble.transactionData.build()) // Set Soroban Data from the simulation
+    .addOperation(Operation.restoreFootprint({})) // Add the RestoreFootprint operation
+    .build(); // Build the transaction
+
+  const simulation: SorobanRpc.Api.SimulateTransactionResponse =
+    await config.rpc.simulateTransaction(restoreTx);
+  const assembledTx = SorobanRpc.assembleTransaction(
+    restoreTx,
+    simulation,
+  ).build();
+
+  const signedTx = new Transaction(
+    await txParams.signerFunction(assembledTx.toXDR()),
+    config.passphrase,
+  );
+
+  console.log("Submitting restoration transaction");
+
+  try {
+    // Submit the transaction to the network
+    const response = await config.rpc.sendTransaction(signedTx);
+    console.log(
+      "Restoration transaction submitted successfully:",
+      response.hash,
+    );
+  } catch (error) {
+    console.error("Failed to submit restoration transaction:", error);
+    throw new Error("Restoration transaction failed");
+  }
+}
+
+import {
+  Account,
+  Keypair,
+  Operation,
+  SorobanDataBuilder,
+  SorobanRpc,
+  TimeoutInfinite,
+  Transaction,
+  TransactionBuilder,
+  scValToNative,
+  xdr,
+} from "@stellar/stellar-sdk";
+```
+
+## Fees and Resource Usage
+
+Soroban smart contracts on Stellar use a multidimensional resource fee model, charging fees for several resource types:
```

Comment:
cross-reference to https://soroban.stellar.org/docs/fundamentals-and-concepts/fees-and-metering or a subsection therein, perhaps?